Entities/Common/Respawning/StandardRespawnCommand.as

```diff
@@ -80,12 +80,12 @@ void onRespawnCommand(CBlob@ this, u8 cmd, CBitStream @params)
 
 		case SpawnCmd::changeClass:
 		{
-			if (getNet().isServer())
+			if (isServer())
 			{
 				// build menu for them
 				CBlob@ caller = getBlobByNetworkID(params.read_u16());
 
-				if (caller !is null && canChangeClass(this, caller))
+				if (caller !is null && !caller.hasTag("dead") && canChangeClass(this, caller))
```

Comment:
It would be wiser to put this check into the canChangeClass bool- rather than in this line

Reply:
Resolved.